content/blog/updating-to-cpp11/index.Rmd

```diff
@@ -0,0 +1,90 @@
+* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [x] Find photo & update yaml metadata
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [x] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+Over the summer I had the pleasure of working with Jim Hester on the [cpp11 package](https://cpp11.r-lib.org/) as a tidyverse summer intern.
+The cpp11 package is a header-only R package that helps R package developers handle R objects with C++ code.
+Its goals and syntax are similar to the excellent Rcpp package.
+During most of my internship, I worked on triaging issues, fixing bugs, and adding new features to cpp11.
+Near the end of the summer, I got to work with Jenny Bryan on converting the [readxl](https://github.com/tidyverse/readxl/pull/659) package from using Rcpp to using cpp11.
+Jim has written a [great post](https://cpp11.r-lib.org/articles/converting.html) about converting packages from Rcpp to cpp11, which I heavily referenced during the process.
+But there were still some challenges I encountered.
+To help others going through a similar process, I wanted to review the workflows and tools we used to make this easier.
+
+To get started, I followed the initial set-up steps outlined in Jim’s article and then recompiled, to confirm it was successful.
+Next, I needed to include either the `cpp11/R.hpp` header or the macros `R_NO_REMAP` and `STRICT_R_HEADERS`.
+I found it easier to include the macros since the header file needs to come before any headers that use Rcpp.
+
+## Step-by-step integration
+
+After I'd set-up my environment it was time to start converting the files.
+Initially, I made the mistake of selecting the most interconnected file in the readxl package.
+Since this file was central to the package, I would have had to convert the entire package over to cpp11 before the compilation errors would stop.
+After Jenny and Jim helped me select a more approachable file, the process was more incremental and I successfully maintained my sanity 🙌.
+
+After I finished converting a file, I recompiled, ran the tests for the readxl package and fixed any compilation errors or test failures.
+Sometimes I had to edit other functions in other files, but I only edited files enough to fix the failures.
+If I did too much it was difficult for me to know what was causing the test failures.
+Another incremental approach I used was to convert one function in a file, and then rerun the tests.
+This approach worked best for larger files.
+Once the tests were passing, I'd commit and push to the PR so that my gracious reviewers could review my changes.
+This helped me maintain continuous integration and also helped my mentors review my code in stages, rather than in one, big batch.
```

Comment:
```suggestion
Once the tests were passing, I'd commit and push to the PR so that my gracious reviewers could review my changes in stages, rather than in one big batch.
This also allowed me to benefit from continuous integration, i.e. my changes were checked on a greater variety of R versions and operating systems, thanks to the checks we run via GitHub Actions.
```
```suggestion
Once the tests were passing, I'd commit and push to the PR so that my gracious reviewers could review my changes.
This helped me maintain continuous integration and also helped my mentors review my code in stages, rather than in one, big batch.
```